custom/icds_reports/utils/aggregation_helpers/agg_child_health.py

```diff
@@ -134,46 +134,55 @@ def aggregation_query(self):
             ('zscore_grading_wfh_recorded_in_month', "SUM(chm.zscore_grading_wfh_recorded_in_month)"),
             ('days_ration_given_child', "SUM(chm.days_ration_given_child)"),
         )
+        query_cols = []
+        for c in columns:
+            if len(c) == 3:
+                name = c[2]
+            else:
+                name = c[0]
+            query_cols.append((name, c[1]))
         return """
-        INSERT INTO "{tablename}" (
-            {columns}
-        ) (SELECT
-            {calculations}
+        CREATE TEMPORARY TABLE "{tmp_tablename}" AS SELECT
+            {cols}
             FROM "{child_health_monthly_table}" chm
             LEFT OUTER JOIN "awc_location" awc_loc ON awc_loc.doc_id = chm.awc_id
             WHERE chm.month = %(start_date)s AND awc_loc.state_id != '' AND awc_loc.state_id IS NOT NULL
             GROUP BY awc_loc.state_id, awc_loc.district_id, awc_loc.block_id, awc_loc.supervisor_id, chm.awc_id,
                      chm.month, chm.sex, chm.age_tranche, chm.caste,
                      coalesce_disabled, coalesce_minority, coalesce_resident
-            ORDER BY awc_loc.state_id, awc_loc.district_id, awc_loc.block_id, awc_loc.supervisor_id, chm.awc_id
-        )
+            ORDER BY awc_loc.state_id, awc_loc.district_id, awc_loc.block_id, awc_loc.supervisor_id, chm.awc_id;
+        INSERT INTO "{tablename}" ({columns}) SELECT * from "{tmp_tablename}";
+        DROP TABLE "{tmp_tablename}";
         """.format(
             tablename=self.tablename,
             columns=", ".join([col[0] for col in columns]),
             calculations=", ".join([col[1] for col in columns]),
+            cols=", ".join(['{} as {}'.format(q, name) for name, q in query_cols]),
             child_health_monthly_table='child_health_monthly',
+            tmp_tablename='tmp_{}'.format(self.tablename)
         ), {
             "start_date": self.month
         }
 
     def update_queries(self):
         yield """
-            UPDATE "{tablename}" agg SET
+        CREATE TEMPORARY TABLE "{tmp_tablename}" AS SELECT
```

Comment:
is this necessary here? neither the table being updated (`agg_child_health....`) and the table being selected from (`awc_location`) are local tables so I don't think the temp table approach is necessary here

Reply:
offlined - `awc_location` is a reference table - some uncertainty about that but ok for now